dvc/remote/gdrive/waitable_lock.py

```diff
@@ -0,0 +1,29 @@
+from time import time, sleep
+from threading import Lock
+
+import zc.lockfile
+
+
+class WaitableLock(object):
+    def __init__(self, lock_file, timeout=5):
+        self.lock_file = lock_file
+        self.timeout = timeout
+        self._thread_lock = Lock()
+        self._lock = None
+
+    def __enter__(self):
+        t0 = time()
+        self._thread_lock.acquire()
+        while time() - t0 < self.timeout:
+            try:
+                self._lock = zc.lockfile.LockFile(self.lock_file)
```

Comment:
is it a regular lockfile or is there something specific? we are using lockfile already somewhere, a different implementation, so do we need zc here? should we specify the dependency explicitly then? Also, what happens if we break the execution in the middle, it will start raising an exception? should we at least explain how to recover from it.

Reply:
@shcheklein We are using zc.lockfile in other places too :) Not sure about the purpose of this lock though, do we actually write to the file it is protecting anywhere @ei-grad ?